### PR TITLE
PR: Fix fileswitcher size during its setup

### LINKED ISSUE #4648: File switcher is changing size when changing from symbols to files
Easy way to reproduce this

1. Press the file switcher button in the file toolbar.
2. Press `Esc` to hide the file switcher.
3. Press the symbol finder button in the file toolbar.
4. Press again the file switcher button.

@dalthviz, besides fixing this error, we have to make sure to remove all text present in the switcher every time we're going to show it again.

### PR DESCRIPTION
**Pull request by dalthviz**
Fixes #4648

### CODE REVIEW

File: spyder/widgets/fileswitcher.py [hunk 1]
```diff
@@ -667,6 +673,7 @@ def setup_symbol_list(self, filter_text, current_path):
         oedata = self.get_symbol_list()
         icons = get_python_symbol_icons(oedata)
 
+        short_paths = shorten_paths(self.paths, self.save_status)
```
Comment (ccordoba12) [inline spyder/widgets/fileswitcher.py]: Please add a comment above this line to explain why this is needed.

File: spyder/widgets/fileswitcher.py [hunk 2]
```diff
@@ -652,7 +658,7 @@ def setup_file_list(self, filter_text, current_path):
             self.set_current_row(self.filtered_path.index(current_path))
         elif self.filtered_path:
             self.set_current_row(0)
-        self.fix_size(short_paths, extra=200)
+        self.fix_size(short_paths, extra=100)
```
Comment (ccordoba12) [inline spyder/widgets/fileswitcher.py]: Why the change from 200 to 100?

File: spyder/widgets/fileswitcher.py [hunk 3]
```diff
@@ -488,6 +491,9 @@ def previous_row(self):
 
     def next_row(self):
         """Select next row in list widget."""
+        if self.mode == self.SYMBOL_MODE:
+            self.select_row(+1)
+            return
```
Comment (ccordoba12) [inline spyder/widgets/fileswitcher.py]: Why is this needed?

File: spyder/widgets/fileswitcher.py (hunk 2 above)
Comment (dalthviz) [inline spyder/widgets/fileswitcher.py]: Since the list for the symbols normally has less width, I was thinking that would be better to reduce the width given by the `extra` param.

File: spyder/widgets/fileswitcher.py (hunk 3 above)
Comment (dalthviz) [inline spyder/widgets/fileswitcher.py]: This is needed since I found a new bug that doesn't allow you to move with the symbol finder. Since is the same element, I thought this could be included in this PR, should I open an issue with a different PR?
